copycat-runtime/src/main/java/org/apache/kafka/copycat/runtime/Coordinator.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package org.apache.kafka.copycat.runtime;
+
+import org.apache.kafka.copycat.util.Callback;
+
+import java.util.Properties;
+
+/**
+ * The Coordinator interface works with coordinators on other workers to manage a set of Jobs.
+ * Each job is a Connector instance with associated config and triggers tasks that are then run
+ * in workers. The assignment and tracking of those tasks in workers is also managed by the
+ * coordinator.
+ */
+public interface Coordinator {
```

Comment:
KIP-26 omitted a lot of implementation details. To be honest, I'm not certain how large this interface will get yet. I think it will have to contain most of the functionality that is exposed via the REST API since that implementation will essentially just be invoking methods on this class, but how large will that end up being.

Originally the thought for this class is that the DistributedCoordinator would be a significantly more complex implementation than the standalone version. However, if we are able to leverage the consumer coordinator code, then it may not even be that complex -- mostly just handles forwarding to the current leader (so, e.g., connector configs are only written by a single worker at any given time).